File: LeetCode/Easy/0145-binary-tree-postorder-traversal/0145-binary-tree-postorder-traversal.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func postorderTraversal(root *TreeNode) []int {
    var ret []int
    postOrder(root, &ret)
    return ret
}

func postOrder(node *TreeNode, container *[]int) {
    if node == nil { return }
    postOrder(node.Left, container)
    postOrder(node.Right, container)
    *container = append(*container, node.Val)
    return
}

File: LeetCode/Easy/0145-binary-tree-postorder-traversal/0145-binary-tree-postorder-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> postorderTraversal(TreeNode* root) {
        vector<int> ret;
        postOrder(root, &ret);
        return ret;
    }
    
    void postOrder(TreeNode* node, vector<int>* container){
        if (!node) return;
        postOrder(node->left, container);
        postOrder(node->right, container);
        container->push_back(node->val);
        return;
    }
};